<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerVirtualRealityModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerVirtualRealityModuleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>367</width>
    <height>632</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>qSlicerVirtualRealityModuleWidget</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="ctkCollapsibleButton" name="ConnectionCollapsibleButton">
     <property name="minimumSize">
      <size>
       <width>300</width>
       <height>0</height>
      </size>
     </property>
     <property name="text">
      <string>Connection</string>
     </property>
     <layout class="QFormLayout" name="formLayout_4">
      <item row="0" column="0">
       <widget class="QLabel" name="XRRuntimeLabel">
        <property name="text">
         <string>Desired XR runtime:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="XRRuntimeComboBox"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="ConnectToHardwareLabel">
        <property name="text">
         <string>Connect to hardware:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <layout class="QHBoxLayout" name="ConnectionLayout">
        <item>
         <widget class="ctkCheckBox" name="ConnectCheckBox"/>
        </item>
        <item>
         <widget class="QLabel" name="ConnectionStatusLabel">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Expanding" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="palette">
           <palette>
            <active>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>0</green>
                <blue>0</blue>
               </color>
              </brush>
             </colorrole>
            </active>
            <inactive>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>255</red>
                <green>0</green>
                <blue>0</blue>
               </color>
              </brush>
             </colorrole>
            </inactive>
            <disabled>
             <colorrole role="WindowText">
              <brush brushstyle="SolidPattern">
               <color alpha="255">
                <red>190</red>
                <green>190</green>
                <blue>190</blue>
               </color>
              </brush>
             </colorrole>
            </disabled>
           </palette>
          </property>
          <property name="text">
           <string/>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="EnableRenderingLabel">
        <property name="text">
         <string>Enable rendering:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="ctkCheckBox" name="RenderingEnabledCheckBox"/>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="DisplayCollapsibleButton">
     <property name="text">
      <string>Display</string>
     </property>
     <layout class="QFormLayout" name="formLayout_3">
      <item row="0" column="0">
       <widget class="QLabel" name="TwoSidedLightingLabel">
        <property name="text">
         <string>Two-sided lighting:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="ctkCheckBox" name="TwoSidedLightingCheckBox"/>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="BackLightsLabel">
        <property name="text">
         <string>Back-lights:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="ctkCheckBox" name="BackLightsCheckBox"/>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="UpdateRateLabel">
        <property name="text">
         <string>Update rate:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="ctkSliderWidget" name="DesiredUpdateRateSlider">
        <property name="toolTip">
         <string>Desired update rate of the view. Higher value reduces time lag but may decrease display quality of volume rendering. It has only effect if volume rendering quality setting is Adaptive.</string>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="maximum">
         <double>120.000000000000000</double>
        </property>
        <property name="value">
         <double>60.000000000000000</double>
        </property>
        <property name="suffix">
         <string> fps</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0" colspan="2">
       <widget class="QPushButton" name="UpdateViewFromReferenceViewCameraButton">
        <property name="text">
         <string>Set virtual reality view to match reference view.</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="ControllerModelsVisibleLabel">
        <property name="text">
         <string>Controllers visible:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="ctkCheckBox" name="ControllerModelsVisibleCheckBox">
        <property name="toolTip">
         <string>If checked, then controllers are displayed in virtual reality view.</string>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="LighthousesVisibleLabel">
        <property name="text">
         <string>Lighthouses visible:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QCheckBox" name="LighthousesVisibleCheckBox">
        <property name="text">
         <string/>
        </property>
        <property name="checked">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="AdvancedCollapsibleButton">
     <property name="text">
      <string>Advanced</string>
     </property>
     <layout class="QFormLayout" name="formLayout_5">
      <item row="0" column="0">
       <widget class="QLabel" name="ReferenceViewLabel">
        <property name="text">
         <string>Reference view:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="qMRMLNodeComboBox" name="ReferenceViewNodeComboBox">
        <property name="nodeTypes">
         <stringlist>
          <string>vtkMRMLViewNode</string>
         </stringlist>
        </property>
        <property name="showChildNodeTypes">
         <bool>false</bool>
        </property>
        <property name="addEnabled">
         <bool>false</bool>
        </property>
        <property name="removeEnabled">
         <bool>false</bool>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="MotionSensitivityLabel">
        <property name="text">
         <string>Motion sensitivity:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="ctkSliderWidget" name="MotionSensitivitySlider">
        <property name="toolTip">
         <string>If no motion of the virtual reality headsets is detected then update rate is decreased to allow higher quality display.</string>
        </property>
        <property name="decimals">
         <number>0</number>
        </property>
        <property name="minimum">
         <double>0.000000000000000</double>
        </property>
        <property name="maximum">
         <double>100.000000000000000</double>
        </property>
        <property name="value">
         <double>50.000000000000000</double>
        </property>
        <property name="suffix">
         <string>%</string>
        </property>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="MagnificationLabel">
        <property name="text">
         <string>Magnification:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <layout class="QHBoxLayout" name="MagnificationLayout">
        <item>
         <widget class="ctkSliderWidget" name="MagnificationSlider">
          <property name="toolTip">
           <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Scale of the world in VR view.&lt;/p&gt;&lt;p&gt;&lt;br/&gt;&lt;/p&gt;&lt;p&gt;1x magnification means that objects appear in real size, 10x means they appear ten times larger.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
          </property>
          <property name="decimals">
           <number>2</number>
          </property>
          <property name="singleStep">
           <double>0.100000000000000</double>
          </property>
          <property name="pageStep">
           <double>0.250000000000000</double>
          </property>
          <property name="minimum">
           <double>-2.000000000000000</double>
          </property>
          <property name="maximum">
           <double>2.000000000000000</double>
          </property>
          <property name="value">
           <double>1.000000000000000</double>
          </property>
          <property name="spinBoxVisible">
           <bool>false</bool>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="MagnificationValueLabel">
          <property name="frameShape">
           <enum>QFrame::StyledPanel</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="text">
           <string>10x</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="MotionSpeedLabel">
        <property name="text">
         <string>Motion speed:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="ctkSliderWidget" name="MotionSpeedSlider">
        <property name="toolTip">
         <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Speed of flying in virtual reality in m/s.&lt;/p&gt;&lt;p&gt;This speed applies in the &amp;quot;physical&amp;quot; coordinate system, which corresponds to physical head and controller movement. This means that if magnification is 10x, then flying will appear 10x slower relative to the scene, but in the physical scale it is constant.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
        </property>
        <property name="decimals">
         <number>1</number>
        </property>
        <property name="singleStep">
         <double>0.500000000000000</double>
        </property>
        <property name="pageStep">
         <double>2.000000000000000</double>
        </property>
        <property name="minimum">
         <double>0.100000000000000</double>
        </property>
        <property name="maximum">
         <double>10.000000000000000</double>
        </property>
        <property name="value">
         <double>1.660000000000000</double>
        </property>
        <property name="suffix">
         <string>m/s</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="ControllerTransformsLabel">
        <property name="text">
         <string>Controller transforms:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="ctkCheckBox" name="ControllerTransformsUpdateCheckBox">
        <property name="toolTip">
         <string>If checked, then position and orientation of each controller will be exposed in the scene as a transform node</string>
        </property>
       </widget>
      </item>
      <item row="6" column="0">
       <widget class="QLabel" name="HMDTransformLabel">
        <property name="text">
         <string>HMD transform:</string>
        </property>
       </widget>
      </item>
      <item row="6" column="1">
       <widget class="QCheckBox" name="HMDTransformCheckBox">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="TrackerTransformsLabel">
        <property name="text">
         <string>Tracker transforms:</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QCheckBox" name="TrackerTransformsCheckBox">
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>0</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCheckBox</class>
   <extends>QCheckBox</extends>
   <header>ctkCheckBox.h</header>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkSliderWidget</class>
   <extends>QWidget</extends>
   <header>ctkSliderWidget.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>qSlicerVirtualRealityModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>ReferenceViewNodeComboBox</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>134</x>
     <y>145</y>
    </hint>
    <hint type="destinationlabel">
     <x>322</x>
     <y>198</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
